core/src/main/scala/cats/instances/eq.scala

```diff
@@ -43,4 +45,25 @@ trait EqInstances extends kernel.instances.EqInstances {
       def product[A, B](fa: Eq[A], fb: Eq[B]): Eq[(A, B)] =
         (left, right) => fa.eqv(left._1, right._1) && fb.eqv(left._2, right._2)
     }
+
+  implicit val catsDeferForEq: Defer[Eq] =
```

Comment:
To fix bincompat these have to be declared as `def`s. It's a bit annoying, but you can cache the singleton instance in an `object` and then refer to it here.